src/databricks/labs/remorph/intermediate/infer_schema.py

```diff
@@ -0,0 +1,57 @@
+import logging
+from pathlib import Path
+
+from databricks.sdk import WorkspaceClient
+
+from databricks.labs.remorph.helpers.file_utils import (
+    get_sql_file,
+    is_sql_file,
+    read_file,
+)
+from databricks.labs.remorph.intermediate.engine_adapter import EngineAdapter
+
+logger = logging.getLogger(__name__)
+
+
+class InferSchema:
+    def __init__(self, ws: WorkspaceClient, source: str, input_path: str | Path):
```

Comment:
```suggestion
    def __init__(self, ws: WorkspaceClient, engine: str):
```

Reply:
Don't tie core logic of how we process queries with how they are stored - we must not have coupling on filesystem and let this core be side-effect free

Reply:
File path / folder path should just be arguments for specialized methods